src/OrchardCore/OrchardCore.Data/OrchardCoreBuilderExtensions.cs

```diff
@@ -66,10 +67,23 @@ public static OrchardCoreBuilder AddDataAccess(this OrchardCoreBuilder builder)
                             var shellOptions = sp.GetService<IOptions<ShellOptions>>();
                             var option = shellOptions.Value;
                             var databaseFolder = Path.Combine(option.ShellsApplicationDataPath, option.ShellsContainerName, shellSettings.Name);
-                            var databaseFile = Path.Combine(databaseFolder, "yessql.db");
                             Directory.CreateDirectory(databaseFolder);
+
+                            var connectionString = shellSettings["ConnectionString"];
+                            if (connectionString.StartsWith("Data Source="))
+                            {
+                                var connectionStringBuilder = new SqliteConnectionStringBuilder(connectionString);
+                                connectionStringBuilder.DataSource = Path.Combine(databaseFolder, connectionStringBuilder.DataSource);
+                                connectionString = connectionStringBuilder.ConnectionString;
+                            }
+                            else
+                            {
+                                var databaseFile = Path.Combine(databaseFolder, "yessql.db");
```

Comment:
This is for backward compatibility, which will make @deanmarcussen happy ;) but I think it should be removed after the `1.0.0` release